router/http/src/main/scala/io/buoyant/router/http/StripHopByHopHeadersFilter.scala

```diff
@@ -0,0 +1,55 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle.http.Fields._
+import com.twitter.finagle.http.{Message, Request, Response}
+import com.twitter.finagle.{Service, ServiceFactory, SimpleFilter, Stack}
+
+object StripHopByHopHeadersFilter {
+
+  object HopByHopHeaders {
+    val Headers = Seq(
+      Connection,
+      ProxyAuthenticate,
+      ProxyAuthorization,
+      Te,
+      Trailer,
+      TransferEncoding,
+      Upgrade
+    )
+
+    def scrub(msg: Message): Unit = {
+      Headers.foreach {
+        case h@Connection =>
+          val headersListedInConnection: Seq[String] = msg.headerMap.remove(h) match {
+            case Some(s) => s.split(",").map(_.trim).filter(_.nonEmpty)
+            case None => Nil
+          }
+          headersListedInConnection.foreach(msg.headerMap.remove(_))
+        case h => msg.headerMap.remove(h)
+      }
+    }
+  }
```

Comment:
Since this is going to run on _all_ requests that traverse http routers, we want to be as conservative as possible about allocation.  And, unfortunately, things like `list.foreach { _ => ... }` incur an allocation _for the anonymous function!_

We can prevent this allocation by "val-ing up" the anonymous function: 
```scala

def scrub(msg: Message): Unit =
  Headers.foreach(scrubHeader)

private[this] val scrubHeader: String => Unit = {
  case h@Connection => ...
  case h => msg.headerMap.remove(h)
}
```

This allows us to only allocate _scrubHeader_ once and not on each invocation of _scrub_.